src/corecel/io/Logger.hh

```diff
@@ -44,33 +49,44 @@
     ::celeritas::world_logger()(CELER_CODE_PROVENANCE, \
                                 ::celeritas::LogLevel::LEVEL)
 
-//---------------------------------------------------------------------------//
 /*!
  * \def CELER_LOG_LOCAL
  *
  * Like \c CELER_LOG but for code paths that may only happen on a single
- * process. Use sparingly.
+ * process or thread. Use sparingly.
  */
 #define CELER_LOG_LOCAL(LEVEL)                        \
     ::celeritas::self_logger()(CELER_CODE_PROVENANCE, \
                                ::celeritas::LogLevel::LEVEL)
 
+// Allow CELER_LOG to be present (but ignored) in device code
+#if CELER_DEVICE_COMPILE
+#    undef CELER_LOG
+#    define CELER_LOG(LEVEL) ::celeritas::detail::NullLoggerMessage()
+#    undef CELER_LOG_LOCAL
+#    define CELER_LOG_LOCAL(LEVEL) ::celeritas::detail::NullLoggerMessage()
+#endif
+
 namespace celeritas
 {
 class MpiCommunicator;
 
 //---------------------------------------------------------------------------//
 /*!
- * Manage logging in serial and parallel.
+ * Create a log message to be printed based on output/verbosity sttings.
  *
  * This should generally be called by the \c world_logger and \c
  * self_logger functions below. The call \c operator() returns an object that
  * should be streamed into in order to create a log message.
  *
  * This object \em is assignable, so to replace the default log handler with a
  * different one, you can call \code
-   world_logger = Logger(celeritas::comm_world(), my_handler);
+   world_logger = Logger(my_handler);
  * \endcode
+ *
+ * When using with MPI, the \c world_logger global objects are different on
+ * each process: rank 0 will have a handler that outputs to screen, and the
+ * other ranks will have a "null" handler that suppresses all log output.
```

Comment:
Is that temporary or a restriction we have to live with?  I.e. with additional work in future PRs can we record the output for the other ranks or is there already another mechanism for us to see error messages that have happened in the other ranks?

Reply:
No, this is just the default behavior for `world_logger`. Anyone can assign a new `Logger` object to it on any process, so you could have it (e.g.) log error messages of a certain level on non-rank-0 nodes.